src/os/bluestore/BlueStore.cc

```diff
@@ -10583,6 +10583,9 @@ void BlueStore::_kv_sync_thread()
 	int r = _balance_bluefs_freespace();
 	ceph_assert(r >= 0);
       }
+      if (bluefs) {
```

Comment:
Suggest to consider another thread for this stuff - kv_sync_thread is a very hot path hence it's better to leave it as simple as possible.
MempoolThread is probably the next choice...  Which should be renamed sooner or later to better reflect its purpose

Reply:
Thanks for advice.

Reply:
Yes, MempoolThread is looking more like PeriodicBackgroundTaskThread. ;)